my/browser/export.py

```diff
@@ -0,0 +1,50 @@
+"""
+Parses browser history using [[http://github.com/seanbreckenridge/browserexport][browserexport]]
+"""
+
+REQUIRES = ["browserexport"]
+
+from my.config import browser as user_config
+from my.core import Paths, dataclass
+
+
+@dataclass
+class config(user_config.export):
+    # path[s]/glob to your backed up browser history sqlite files
+    export_path: Paths
+
+
+from pathlib import Path
+from typing import Iterator, Sequence, List
+
+from my.core import Stats, get_files, LazyLogger
+from my.core.common import mcachew
+
+from browserexport.merge import read_and_merge, Visit
+
+from .common import _patch_browserexport_logs
+
+
+logger = LazyLogger(__name__, level="warning")
+
+_patch_browserexport_logs()
+
+
+# all of my backed up databases
+def inputs() -> Sequence[Path]:
+    return get_files(config.export_path)
+
+
+def _cachew_depends_on() -> List[str]:
+    return [str(f) for f in inputs()]
```

Comment:
@seanbreckenridge do you remember why are we doing it here instead of just passing `depends_on=inputs` to cachew? just tried it and seems to work :thinking: maybe cachew didn't support it back then?

Reply:
yeah I dont remember exactly why but I think it had to do with testability/name clashes based on a different name for cachew.

I dont think I fully understood it at the time (may still not), I had this helper type `InputSource` so that mypy was happy:

https://github.com/seanbreckenridge/HPI/blob/master/my/utils/input_source.py

So that I could set the default for the `from_paths` in the function signature itself, but still allow the user to override by passing a function which returned a list of paths as inputs:

https://github.com/seanbreckenridge/HPI/blob/aa69d73cc71017b64287a04d917456da8fe5105a/my/bash.py#L64-L70

Something like:

```
from more_itertools import ilen
from my.bash import history, inputs

print(ilen(history(from_paths=lambda: tuple(inputs()[0:1]))))
```

would sort of obviously invalidate the cache, but yeah...

```
 python3 test.py
[INFO    2023-11-06 16:31:38,732 my.bash __init__.py:784 ] [my.bash:history] wrote   59 objects to   cachew (sqlite:/home/sean/.cache/cachew/my.bash:history)
59
```

Not really sure how important that is, someone could always override the inputs by patching their config (having an example of this in the docs might be nice?), I havent really ever found myself specifying a custom function to the inputs so its probably not a feature worthwhile keeping around

you can change it to whatever makes sense for you

Reply:
I cant reproduce the error anymore but I remember having to rename stuff here specifically:

https://github.com/seanbreckenridge/HPI/blob/aa69d73cc71017b64287a04d917456da8fe5105a/my/bash.py#L64-L70

from `from_paths` to `for_paths` in the cachew helper function. otherwise there was some hard to debug error I couldnt deal with. doesnt seem to happen anymore

Reply:
yeah good point about documenting patching the config! I was kinda reluctant because it's kinda hacky, but seems like it's working as intended so far
https://github.com/karlicoss/promnesia/blob/master/src/promnesia/sources/browser.py#L53-L67